src/server/player_sao.cpp

```diff
@@ -519,12 +519,15 @@ void PlayerSAO::rightClick(ServerActiveObject *clicker)
 
 void PlayerSAO::setHP(s32 target_hp, const PlayerHPChangeReason &reason, bool from_client)
 {
+	target_hp = rangelim(target_hp, -U16_MAX, U16_MAX);
```

Comment:
This is here to prevent potential overflowing of s32 type.
But I can replace it with std::max<s32>(target_hp, -U16_MAX).
So if `target_hp` is near the `S32_MIN` value, it will not overflow when hp is subtracted.